apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/configutation/AuthConfiguration.java

```diff
@@ -0,0 +1,149 @@
+      singleSignOutFilter.addUrlPatterns("/*");
+      return singleSignOutFilter;
+    }
+
+    @Bean
+    public FilterRegistrationBean authenticationFilter(){
+      FilterRegistrationBean casFilter = new FilterRegistrationBean();
+
+      Map<String, String> filterInitParam = new HashMap();
+      filterInitParam.put("redisClusterName", "casClientPrincipal");
+      filterInitParam.put("serverName", serverConfigRepository.findByKey("serverName").getValue());
+      filterInitParam.put("casServerLoginUrl", serverConfigRepository.findByKey("casServerLoginUrl").getValue());
+
+      casFilter.setInitParameters(filterInitParam);
+      casFilter.setFilter(filter("org.jasig.cas.client.authentication.AuthenticationFilter"));
+      casFilter.addUrlPatterns("/*");
+
+      return casFilter;
+    }
+
+    @Bean
+    public FilterRegistrationBean casValidationFilter(){
+      FilterRegistrationBean casValidationFilter = new FilterRegistrationBean();
+      Map<String, String> filterInitParam = new HashMap();
+      filterInitParam.put("casServerUrlPrefix", serverConfigRepository.findByKey("casServerUrlPrefix").getValue());
+      filterInitParam.put("serverName", serverConfigRepository.findByKey("serverName").getValue());
+      filterInitParam.put("encoding", "UTF-8");
+      filterInitParam.put("useRedis", "true");
+      filterInitParam.put("redisClusterName", "casClientPrincipal");
+
+      casValidationFilter.setFilter(filter("org.jasig.cas.client.validation.Cas20ProxyReceivingTicketValidationFilter"));
+      casValidationFilter.setInitParameters(filterInitParam);
+      casValidationFilter.addUrlPatterns("/*");
+
+      return casValidationFilter;
+
+    }
+
+
+
+    @Bean
+    public FilterRegistrationBean assertionHolder(){
+      FilterRegistrationBean assertionHolderFilter = new FilterRegistrationBean();
+
+      assertionHolderFilter.setFilter(filter("org.jasig.cas.client.util.AssertionThreadLocalFilter"));
+      assertionHolderFilter.addUrlPatterns("/*");
+
+      return assertionHolderFilter;
+    }
+
+    @Bean
+    public CtripUserInfoHolder ctripUserInfoHolder(){
+      return new CtripUserInfoHolder();
+    }
+
+    private Filter filter(String className){
+      Class clazz = null;
+      try {
+        clazz = Class.forName(className);
+        Object obj = clazz.newInstance();
+        return (Filter) obj;
+      } catch (Exception e) {
+        throw new RuntimeException("instance filter fail", e);
+      }
+
+    }
+
+    private EventListener listener(String className){
+      Class clazz = null;
+      try {
+        clazz = Class.forName(className);
+        Object obj = clazz.newInstance();
+        return (EventListener) obj;
+      } catch (Exception e) {
+        throw new RuntimeException("instance listener fail", e);
+      }
+    }
+  }
+
+  /**
+   * 默认实现
+   */
+  @Configuration
+  static class NotCtripProfileConfiguration{
```

Comment:
Could name this class as DefaultAuthConfiguration
Should we mark this with @Profile ?